### ISSUE TITLE
Update NEO-2 docs

### ISSUE THREAD

**Issue by krystophny** [itpplasma/NEO-2#16]
Radically reduce outdated information

- Build is now much simpler
- Half-finished things can either be completed or deleted

**Comment by krystophny** [comment]
*Automated triage pass (Claude), grounded in the repository at `80a6032`. Verify before acting.*

Still valid. Both checkboxes are unfinished at `main` HEAD, and the doc tree has drifted further since this was filed in Dec 2024 (build overhaul `927eb1b`/`62dac03`, MyMPILib move `8bedccc`, libneo decomposition `ba465f6`). Below is a concrete inventory of what is stale, plus a spec.

## Purpose

`DOC/` is the only place a new user or student learns how to drive NEO-2 beyond "run `make`". Today it actively misleads: `Build.md` describes a build system that was replaced in Jan 2025, `DOC/UserDoc/main.tex` references a bibliography file that does not exist in the repository, and two chapter files are orphaned from the document. A person following `Build.md` literally will `mkdir Build-Debug && cmake ..` and fail. The second item is a license-hygiene obligation: `cmake/UseLATEX.cmake` and `cmake/FindSuiteSparse.cmake` are both third-party BSD 3-Clause files vendored into an MIT repository, and there is no repo-level acknowledgement of either.

## Current state

**Build documentation is contradicted by the actual build.** `Readme.md:17-34` and `Makefile:18-31` are authoritative: `make` drives `cmake --preset default` (`Makefile:21`), `LIBNEO_REF` selects the libneo revision (`Makefile:21`, `Readme.md:25-29`), and the binaries are `neo_2_ql.x` / `neo_2_par.x` (`Readme.md:33-34`, set at `NEO-2-QL/CMakeLists.txt:7` and `NEO-2-PAR/CMakeLists.txt:7`). `Build.md` still says:

```
Build.md:25   * mc_efit
Build.md:38   ProjectConfig.cmake.in (should be made configurable) in the COMMON/
Build.md:71   create it by 'make coding_style_safepdf'.
Build.md:78   directory Build-Debug (other ones, like Build-Release, might be created
Build.md:86   finished, the result will be an executable named 'neo_2.x'.
```

`mc_efit` was replaced by `magfie` (`bd3a4b2`, "Replace MC_EFIT by magfie"); `ProjectConfig.cmake.in` is no longer the configuration surface; there is no `neo_2.x` target anywhere in the tree (`git grep -n 'neo_2\.x' -- '*CMakeLists.txt'` is empty); `NEO-2-PAR/Build-Debug/` and `NEO-2-PAR/Build-Release/` contain only a leftover `BuildConfig.cmake.in`.

**The LaTeX manual has two hard build hazards.** `DOC/UserDoc/main.tex`:

```latex
40  \addbibresource[]{eccd_ref.bib}
50  \usepackage{../../../libneo/doc/shared/Commands}
```

`find . -name eccd_ref.bib` returns nothing — the file is not in the repository, so `biber` has no resource. Line 50 hard-codes a relative path to a sibling **libneo working copy**; `doc/shared/Commands.sty` does exist inside libneo, but that path is not what `FetchContent` produces, so the doc build only works on a developer machine that happens to have libneo cloned next to NEO-2. The `DOC` subdirectory is only added when `LATEX_FOUND AND DOXYGEN_FOUND AND NOT DOXYGEN_DOT_EXECUTABLE STREQUAL ""` (`CMakeLists.txt:126`), and then as `EXCLUDE_FROM_ALL` (`CMakeLists.txt:128`), so this breakage is invisible in CI.

**Two chapters are orphaned.** `DOC/UserDoc/Notes.tex` and `DOC/UserDoc/nbi.tex` exist but are neither `\include`d in `main.tex:62-94` nor listed in the `INPUTS` block of `DOC/CMakeLists.txt:38-48` (`grep -n 'Notes\|nbi' DOC/UserDoc/main.tex DOC/CMakeLists.txt` returns nothing). `DOC/UserDoc/MyMPILib.tex` was added by `8bedccc` ("Move MyMPILib into NEO-2") and deleted again by `824ea2b` ("Keep MyMPILib move focused"), leaving `DOC/UserDoc/Building.tex:38-40` as the only trace ("Now included directly in NEO-2") with no pointer to `COMMON/MyMPILib/` or `TEST/mympilib/`.

**Dangling references to a non-existent document.** `DOC/UserDoc/Introduction.tex:18-21` and `Build.md:71` both tell the reader to build a coding-style document with `make coding_style_safepdf`. `find . -iname '*coding*' -not -path './.git/*'` returns nothing — the source is not in this repository. `CLAUDE.md` is the de facto style reference now.

**Half-finished stubs.** `DOC/UserDoc/Bugs.tex` is 5 lines of chapter preamble with no sections. `DOC/UserDoc/Todo.tex:19` is a literal `\section{}`. `DOC/UserDoc/Building.tex:32` and `:35` are `\subsection{ATLAS}` and `\subsection{OpenBLAS}` with no body. `DOC/UserDoc/Input.tex:10` and `:13` are `\section{PAR}` and `\section{QL}` with no body. Unresolved `(?)` markers sit at `DOC/UserDoc/Running.tex:86` ("depends only on (?)") and `DOC/UserDoc/Programflow.tex:28`.

**Condor-era content.** `DOC/UserDoc/Running.tex:93-105` is an HTCondor primer linking `http://research.cs.wisc.edu/htcondor/manual/`; further condor passages sit at `Running.tex:254,268,270,294,296,303,306,307`. Two of the three DOC submit files invoke the dead binary: `DOC/condor_submit_example:2` and `DOC/condor_submit_failed.template:18` both run `./neo_2.x`. (`DOC/condor_submit_all_reconstructions_example` runs `./NEO2-all.sh`, which does exist at `ShellScripts/NEO2-all.sh` — it is Condor-era but not broken in the same way.) `DOC/slurm.marconi.example:52,57,60,64` runs `./neo_2.x`. `DOC/ExtraDocuments/neo-2-par-eccd_runs.tex:290,297,313,321,579` documents `ProjectConfig.cmake.in`, `cd Build-Debug`, and `cp $NEO2PATH/NEO-2-PAR/Build/neo_2.x .`; `neo-2-ql-demo_runs.tex:281` repeats the `ProjectConfig.cmake.in` line. The scripts these sections drive do still exist (`ShellScripts/run_condor.sh`, `ShellScripts/switch_reconstruction.sh`, `PythonScripts/scan_nonhdf5_tools.py`); whether they are still used is a question for the maintainers, not something the tree answers.

**`CHANGELOG`** was last touched by `8d9c7fc` (2024-11-08), predating every refactor listed above.

**Third-party licenses.** Two vendored CMake modules carry upstream BSD 3-Clause headers:

- `cmake/UseLATEX.cmake:1-37` — version 2.4.6, Kenneth Moreland `<kmorel@sandia.gov>`, "Copyright 2004, 2015 Sandia Corporation" (`:6`), "This software is released under the BSD 3-Clause License" (`:10`), conditions at `:12-19`. Pulled in at `DOC/CMakeLists.txt:34` (`include(UseLATEX)`), and `DOC/CMakeLists.txt:33` already admits "Not from us, but added to the repository for easy acess."
- `cmake/FindSuiteSparse.cmake:1-30` — from Ceres Solver, "Copyright 2013 Google Inc. All rights reserved.", BSD 3-Clause conditions, author Alex Stewart.

`find . -iname '*licen[sc]e*' -not -path './.git/*'` finds only `./LICENSE` and `./python/LICENSE`, both plain MIT ("Copyright (c) 2024 Plasma Physics at ITPcp, TU Graz"), neither with a third-party section. The in-file notices satisfy BSD-3 clause 1, but nothing at repository level tells a redistributor that part of the tree is not MIT.

## Reproduction / evidence

```bash
cd NEO-2   # at 80a6032

# 1. Build.md instructions do not correspond to any target
git grep -n 'neo_2\.x' -- CMakeLists.txt '*/CMakeLists.txt'   # -> no hits
ls -A NEO-2-PAR/Build-Debug/                                  # -> BuildConfig.cmake.in only

# 2. The manual references a missing bibliography and an out-of-tree package
grep -n 'addbibresource\|usepackage{\.\./' DOC/UserDoc/main.tex   # -> lines 40 and 50
find . -name 'eccd_ref.bib' -not -path './.git/*'                 # -> nothing

# 3. Orphaned chapters
grep -n 'Notes\|nbi' DOC/UserDoc/main.tex DOC/CMakeLists.txt   # -> no hits

# 4. Referenced coding-style document does not exist
find . -iname '*coding*' -not -path './.git/*'                 # -> nothing

# 5. Third-party headers present, no repo-level notice
grep -n 'BSD 3-Clause' cmake/UseLATEX.cmake cmake/FindSuiteSparse.cmake
find . -iname '*licen[sc]e*' -not -path './.git/*'             # -> ./LICENSE ./python/LICENSE

# 6. Doc build (needs latex + biber + doxygen + graphviz dot installed)
make doc
```

Expected after this issue is done: `make doc` produces the user manual PDF on a machine with only NEO-2 checked out (no sibling libneo working copy), and `Build.md` either agrees with `Readme.md` or no longer exists.

Observed today: `make doc` cannot succeed from a clean single-repo checkout because of `main.tex:50`; and `Build.md` documents a workflow removed by `927eb1b`/`62dac03`.

## Behaviour (BDD)

```gherkin
Scenario: Documentation build from a clean single-repo checkout
  Given a fresh clone of NEO-2 at main with no sibling libneo working copy
    And latex, biber, doxygen and graphviz dot installed
  When I run "make doc"
  Then the build succeeds with exit status 0
    And the user manual PDF exists under the build tree
    And no log line matches "eccd_ref.bib" or "File `../../../libneo"

Scenario: Build instructions match the real build system
  Given the repository at main
  When I run: git grep -nE 'Build-Debug|ProjectConfig\.cmake\.in|neo_2\.x|mc_efit|coding_style_safepdf' -- Build.md Readme.md DOC
  Then the command exits non-zero with no output, except for paths under DOC/Historical/
    And Readme.md remains the single build reference, naming "make", "make CONFIG=Debug",
        "make LIBNEO_REF=<ref>", "neo_2_ql.x" and "neo_2_par.x"

Scenario: No orphaned chapter files
  Given DOC/UserDoc contains Notes.tex and nbi.tex
  When I list every .tex under DOC/UserDoc except main.tex
  Then each listed file is named both in an \include line of DOC/UserDoc/main.tex
       and in the INPUTS list of DOC/CMakeLists.txt, or has been deleted

Scenario: No bodiless headings remain
  Given DOC/UserDoc/Todo.tex line 19 is "\section{}"
    And DOC/UserDoc/Building.tex lines 32 and 35 are \subsection{ATLAS} and \subsection{OpenBLAS} with no body
    And DOC/UserDoc/Input.tex lines 10 and 13 are \section{PAR} and \section{QL} with no body
  When the cleanup is complete
  Then "git grep -n '\\\\section{}' DOC/UserDoc" returns nothing
    And each of those five headings either has at least one non-blank, non-comment line
        before the next heading, or has been removed

Scenario: Third-party licenses are acknowledged
  Given cmake/UseLATEX.cmake carries a BSD 3-Clause header (Kenneth Moreland, Sandia Corporation, version 2.4.6)
    And cmake/FindSuiteSparse.cmake carries a BSD 3-Clause header (Ceres Solver, Google Inc. 2013)
  When I read THIRD_PARTY_LICENSES.md at the repository root
  Then it has one entry per file, each naming the path, the upstream project and the license
    And the UseLATEX entry gives version 2.4.6 and upstream https://gitlab.kitware.com/kmorel/UseLATEX
    And LICENSE contains a pointer line referring to THIRD_PARTY_LICENSES.md
    And "git diff HEAD -- cmake/UseLATEX.cmake cmake/FindSuiteSparse.cmake" is empty

Scenario: Legacy cluster material is either current or gone
  Given DOC/condor_submit_example:2 and DOC/slurm.marconi.example:52 invoke "./neo_2.x"
  When the cleanup is complete
  Then either those files are deleted together with the Condor sections of DOC/UserDoc/Running.tex
    Or they invoke "./neo_2_par.x" / "./neo_2_ql.x" and are reachable from a section of
       Running.tex that states which cluster they were last validated on
```

## Implementation plan

Split this into three atomic PRs — the license one is trivial and should not wait for the prose work.

**PR 1 — license (small, independent)**

1. Create `THIRD_PARTY_LICENSES.md` at the repo root with two entries:
   - `cmake/UseLATEX.cmake` — version 2.4.6, Kenneth Moreland (Sandia Corporation), upstream `https://gitlab.kitware.com/kmorel/UseLATEX`, BSD 3-Clause, followed by the verbatim header text from `cmake/UseLATEX.cmake:6-37`.
   - `cmake/FindSuiteSparse.cmake` — from Ceres Solver, Copyright 2013 Google Inc., author Alex Stewart, BSD 3-Clause, followed by the verbatim header text from `cmake/FindSuiteSparse.cmake:2-28`.
2. Edit `LICENSE`: append below the MIT text — "Portions of this repository are third-party code under different licenses; see THIRD_PARTY_LICENSES.md."
3. Edit `Readme.md`: add a `## License` section stating MIT plus the pointer to `THIRD_PARTY_LICENSES.md`.
4. Do **not** edit either vendored file — clause 1 of BSD-3 requires the headers be retained as-is.

**PR 2 — build documentation**

5. Delete `Build.md`. Its substantive statements are all wrong (see Current state) and `Readme.md:17-34` covers the same ground correctly.
6. Rewrite `DOC/UserDoc/Building.tex` as a thin chapter: point to `Readme.md` for the canonical instructions, keep only the dependency prose, and update it — drop `\subsection{SuperLu}` (`Building.tex:22-24`, which already says the branch was removed in December 2024), fill or delete the bodiless `\subsection{ATLAS}` / `\subsection{OpenBLAS}` (`Building.tex:32,35`), and correct `\subsection{MPI/MyMPILib}` (`Building.tex:38-40`) to name `COMMON/MyMPILib/`, the CMake target `NEO2MyMPILib` (`COMMON/MyMPILib/CMakeLists.txt:16`), and the test at `TEST/mympilib/test_mympilib.f90`. Add SuiteSparse auto-build (`cmake/suitesparse.cmake`), FGSL (`cmake/fgsl.cmake`), NetCDF, and the granular libneo targets from `COMMON/CMakeLists.txt:114-119` (`LIBNEO::magfie`, `LIBNEO::odeint`, `LIBNEO::hdf5_tools`, `LIBNEO::util`).
7. Edit `DOC/UserDoc/Introduction.tex:18-21`: remove the `coding_style_safepdf` paragraph and replace it with a reference to `CLAUDE.md`.
8. Fix `DOC/UserDoc/main.tex:40` — either add the missing `eccd_ref.bib` under `DOC/UserDoc/` (if the manual actually cites anything) or, if nothing cites, delete the whole biblatex block at `main.tex:23-44`.
9. Fix `DOC/UserDoc/main.tex:50` — vendor a minimal `DOC/UserDoc/Commands.sty` into this repo and use `\usepackage{Commands}`, or make the path configurable via a CMake-substituted variable in `DOC/CMakeLists.txt`. The vendored copy is simpler and removes the sibling-checkout assumption; if it is copied from libneo, add libneo's own license note to `THIRD_PARTY_LICENSES.md` from PR 1.
10. Edit `DOC/CMakeLists.txt` / `.github/workflows/unit-tests.yml`: after step 9, add a smoke check so this cannot rot silently — a job that installs `texlive-latex-recommended`, `biber` and `doxygen` and runs the `doc` target, gated so it does not fail the build when those are absent.

**PR 3 — content pruning**

11. `DOC/UserDoc/Running.tex`: delete `\section{Condor}` (lines 93-105) and the condor passages at 254-307; replace with a short "Cluster runs" section pointing at `DOC/slurm.marconi.example`. Resolve or delete the `(?)` passage at lines 85-89.
12. `DOC/`: delete `condor_submit_example`, `condor_submit_all_reconstructions_example`, `condor_submit_failed.template` unless someone still uses HTCondor; update `slurm.marconi.example:52,57,60,64` to `./neo_2_par.x`.
13. `DOC/ExtraDocuments/neo-2-par-eccd_runs.tex` and `neo-2-ql-demo_runs.tex`: these are run reports, not manuals. Either move them under a clearly-labelled `DOC/Historical/` with a header stating "describes the pre-2025 build; retained for reference", or update the build sections (`neo-2-par-eccd_runs.tex:290,297,313,321,579`; `neo-2-ql-demo_runs.tex:281`). Do not leave them as-is under the same roof as current instructions. If they are moved, update their `add_latex_document` entries at `DOC/CMakeLists.txt:50-51`.
14. `DOC/UserDoc/Notes.tex` and `nbi.tex`: decide per file. `nbi.tex` documents live code — `lsw_nbi` really is in namelist `/collision/` (`NEO-2-QL/neo2.f90:221`, alongside `m_nbi`, `T_nbi`) and module `collop_nbi` is `COMMON/collop_nbi.f90` — so fold it into `DOC/UserDoc/Theory.tex` or add it to `main.tex` and to the `INPUTS` list at `DOC/CMakeLists.txt:38-48`. `Notes.tex` is scratch — either promote the `source_vector` indexing note (`Notes.tex:13-31`) into `Programflow.tex` and delete the rest, or delete the file.
15. `DOC/UserDoc/Bugs.tex` (no content) and `DOC/UserDoc/Todo.tex` (`\section{}` at line 19): delete both files, remove the `\include` lines at `main.tex:90,94` and the `UserDoc/Bugs.tex` / `UserDoc/Todo.tex` entries at `DOC/CMakeLists.txt:39,47`. Open GitHub issues instead — that is what the tracker is for.
16. `DOC/UserDoc/Input.tex:10,13`: fill the bodiless `\section{PAR}` / `\section{QL}` with a one-paragraph pointer each to `DOC/neo2.in.par-full` / `DOC/neo2.in.ql-full` and `DOC/neo.in.par-full` / `DOC/neo.in.ql-full`, and add the booz_xform NetCDF input path (`INP_SWI_BOOZMN = 10` at `COMMON/neo_control.f90:11`, dispatched to `neo_read_boozmn` at `COMMON/neo_sub.f90:944-945`, added by `9b9535b`) which is currently undocumented in the manual.
17. `CHANGELOG` (last touched `8d9c7fc`, 2024-11-08): either add entries for the 2025-2026 refactors or add a header line stating that `git log` and the date-based tags are the record and the file is frozen.

No sibling-repo change is required, with one caveat: step 9 touches material copied from libneo (`doc/shared/Commands.sty`) — if you vendor it, no libneo PR is needed; if you instead want libneo to install it as a data file, that becomes a libneo PR that must merge first.

## Acceptance criteria

- [ ] `THIRD_PARTY_LICENSES.md` exists at the repo root and has an entry for both `cmake/UseLATEX.cmake` and `cmake/FindSuiteSparse.cmake`, each reproducing its BSD 3-Clause text; `LICENSE` and `Readme.md` point to it.
- [ ] `git diff HEAD -- cmake/UseLATEX.cmake cmake/FindSuiteSparse.cmake` is empty.
- [ ] `git grep -nE 'Build-Debug|ProjectConfig\.cmake\.in|neo_2\.x|mc_efit|coding_style_safepdf' -- Readme.md DOC ':(exclude)DOC/Historical'` returns no hits, and `Build.md` is either deleted or free of the same patterns.
- [ ] `find . -name 'eccd_ref.bib' -not -path './.git/*'` finds the file, or `grep -n addbibresource DOC/UserDoc/main.tex` returns nothing.
- [ ] `grep -n 'usepackage{\.\./' DOC/UserDoc/main.tex` returns nothing.
- [ ] On a clean clone with **no sibling libneo directory**, `make doc` exits 0 and produces the user manual PDF under `build/DOC/`.
- [ ] Every `.tex` under `DOC/UserDoc/` other than `main.tex` is named both in an `\include` line of `main.tex` and in the `INPUTS` list of `DOC/CMakeLists.txt`, or has been deleted.
- [ ] `git grep -n '\\section{}' DOC/UserDoc` returns nothing, and none of `Building.tex` ATLAS/OpenBLAS or `Input.tex` PAR/QL is left as a heading with no body.
- [ ] `make && make test` still passes — the ctest suite (`boozmn_read_test`, `nrutil_test`, `er_rotation_test`, `spline_sparse_assembly_test` at `TEST/CMakeLists.txt:13,27,43,56`; `test_mympilib` at `TEST/mympilib/CMakeLists.txt:7`) is unaffected.
- [ ] The suite in `.github/workflows/test-on-pr.yml` still passes; no Fortran source is touched by any of these PRs, so a failure there indicates an accidental non-doc edit.

## References

No specific literature reference; this is documentation and license hygiene internal to the code. For the license side the governing texts are the BSD 3-Clause headers carried in `cmake/UseLATEX.cmake:6-37` (upstream `https://gitlab.kitware.com/kmorel/UseLATEX`) and `cmake/FindSuiteSparse.cmake:2-28` (upstream Ceres Solver).

## Scope and risk

Three small PRs; none touches Fortran, so the risk is confined to the `doc` target, which is already `EXCLUDE_FROM_ALL` (`CMakeLists.txt:128`) and untested in CI — which is precisely why it broke. The real risk is deleting the `ExtraDocuments` run reports: they contain workflow knowledge (multi-species scans, reconstruction stages) that exists nowhere else, so archive rather than delete. If other repositories in the org vendor third-party CMake modules, the same `THIRD_PARTY_LICENSES.md` convention is worth reusing for consistency.
